packages/features/form-builder/FormBuilder.tsx

```diff
@@ -418,398 +242,430 @@ export const FormBuilder = function FormBuilder({
+  onOpenChange: (isOpen: boolean) => void;
+  handleSubmit: SubmitHandler<RhfFormField>;
+}) {
+  const { t } = useLocale();
+  const fieldForm = useForm<RhfFormField>({
+    defaultValues: dialog.data || {},
+    // resolver: zodResolver(fieldSchema),
+  });
+  useEffect(() => {
+    if (!fieldForm.getValues("type")) {
+      return;
     }
-  | {
-      value: boolean;
-      setValue: (value: boolean) => void;
-    };
 
-export const ComponentForField = ({
-  field,
-  value,
-  setValue,
-  readOnly,
-}: {
-  field: Omit<RhfFormField, "editable" | "label"> & {
-    // Label is optional because radioInput doesn't have a label
-    label?: string;
-  };
-  readOnly: boolean;
-} & ValueProps) => {
-  const fieldType = field.type || "text";
-  const componentConfig = Components[fieldType];
-
-  const isValueOfPropsType = (val: unknown, propsType: typeof componentConfig.propsType) => {
-    const propsTypeConditionMap = {
-      boolean: typeof val === "boolean",
-      multiselect: val instanceof Array && val.every((v) => typeof v === "string"),
-      objectiveWithInput: typeof val === "object" && val !== null ? "value" in val : false,
-      select: typeof val === "string",
-      text: typeof val === "string",
-      textList: val instanceof Array && val.every((v) => typeof v === "string"),
-    } as const;
-    if (!propsTypeConditionMap[propsType]) throw new Error(`Unknown propsType ${propsType}`);
-    return propsTypeConditionMap[propsType];
-  };
+    const variantsConfig = getVariantsConfig({
+      type: fieldForm.getValues("type"),
+      variantsConfig: fieldForm.getValues("variantsConfig"),
+    });
 
-  // If possible would have wanted `isValueOfPropsType` to narrow the type of `value` and `setValue` accordingly, but can't seem to do it.
-  // So, code following this uses type assertion to tell TypeScript that everything has been validated
-  if (value !== undefined && !isValueOfPropsType(value, componentConfig.propsType)) {
-    throw new Error(
-      `Value ${value} is not valid for type ${componentConfig.propsType} for field ${field.name}`
-    );
+    // We need to set the variantsConfig in the RHF instead of using a derived value because RHF won't have the variantConfig for the variant that's not rendered yet.
+    fieldForm.setValue("variantsConfig", variantsConfig);
+  }, [fieldForm]);
+  const isFieldEditMode = !!dialog.data;
+  const fieldType = fieldTypesConfigMap[fieldForm.watch("type") || "text"];
+
+  const variantsConfig = fieldForm.watch("variantsConfig");
+
+  const fieldTypes = Object.values(fieldTypesConfigMap);
+
+  return (
+    <Dialog open={dialog.isOpen} onOpenChange={onOpenChange}>
+      <DialogContent className="max-h-none p-0" data-testid="edit-field-dialog">
+        <div className="h-auto max-h-[85vh] overflow-auto px-8 pb-7 pt-8">
+          <DialogHeader title={t("add_a_booking_question")} subtitle={t("form_builder_field_add_subtitle")} />
```

Comment:
`form_builder_field_add_subtitle` is missing in common.json

Reply:
I had missed pushing the last commit. Thanks @CarinaWolli as always for catching bugs 🙏